tool/tsh/tsh.go

```diff
@@ -2117,3 +2122,21 @@ func readClusterFlag(cf *CLIConf, fn envGetter) {
 // envGetter is used to read in the environment. In production "os.Getenv"
 // is used.
 type envGetter func(string) string
+
+func handleUnimplementedError(ctx context.Context, cf CLIConf) error {
+	const (
+		errMsgFormat         = "This server does not implement this feature yet. Likely the client version you are using is newer than the server. The server version: %v, the client version: %v. Please upgrade the server or downgrade the client."
+		unknownServerVersion = "unknown"
+	)
+	tc, err := makeClient(&cf, false)
+	if err != nil {
+		log.Warningf("failed to create client: %v", err)
```

Comment:
A couple of stylistic notes specific to how we usually log errors in teleport:

* Each log message we try to make a full sentence starting with a capital letter and ending with a dot.
* Use WithError(err) to log errors (which will dump stack trace as well).

So here and below:

```go
log.WithError(err).Warning("Failed to create client.")
```